newhelm/runners/simple_test_runner.py

```diff
@@ -74,7 +74,8 @@ def run_prompt_response_test(
         if max_test_items < len(test_items):
             rng = random.Random()
             rng.seed(0)
-            test_items = rng.sample(test_items, max_test_items)
+            rng.shuffle(test_items)
```

Comment:
I believe the current logic is already stable. I copied just your tests into my local repo and they passed. I also wrote a little simulator and tested it out with a bunch of values and it was stable.

That said, I think the current implementation is only stable due to implementation details of the `sample` method, whereas your implementation only relies on interface guarantees, so its probably an improvement.

Reply:
Ah, fair enough. You're right, it works the same way for me. I guess the slowness I was seeing on a small increment to `max_test_items` must have been about the cache issues. In that case I don't care about the implementation, but we should keep the test.